src/solver/hydro/management/management.cpp

```diff
@@ -338,6 +326,7 @@ bool HydroManagement::checkMinGeneration(uint numSpace)
         uint z = area.index;
         const auto& ptchro = NumeroChroniquesTireesParPays[numSpace][z];
         auto tsIndex = (uint)ptchro.Hydraulique;
+        auto tsIndexPowerCredits = ptchro.HydrauliquePowerCredits;
```

Comment:
**Important (business rule)** 
So, if I understand correctly : 
- on one hand we have to choose **one** TS number for **levels** and **min generation** (which is a min generation **power** by the way).
- on the other hand, we have to choose **one** TS number for **max generation and pumping power** .

Shouldn't all hourly powers, min or max, go together, and levels stay apart ?

Reply:
Yes we had the same opinion. But Paul from your R&D department explained why it should be like the way we implemented it. So that there is no loss of information in retelling the explanations, best to contact him for clarification.

Reply:
ok. We close this conversation.